pkg/alerts/nrql_conditions.go

```diff
@@ -21,10 +21,62 @@ type AlertsNrqlConditionExpiration struct {
 // AlertsNrqlConditionSignal - Configuration that defines the signal that the NRQL condition will use to evaluate.
 // nolint:revive
 type AlertsNrqlConditionSignal struct {
-	AggregationWindow *int              `json:"aggregationWindow,omitempty"`
-	EvaluationOffset  *int              `json:"evaluationOffset,omitempty"`
-	FillOption        *AlertsFillOption `json:"fillOption"`
-	FillValue         *float64          `json:"fillValue"`
+	AggregationWindow *int                     `json:"aggregationWindow,omitempty"`
+	EvaluationOffset  *int                     `json:"evaluationOffset,omitempty"`
+	FillOption        *AlertsFillOption        `json:"fillOption"`
+	FillValue         *float64                 `json:"fillValue"`
+	AggregationMethod *AlertsAggregationMethod `json:"aggregationMethod,omitempty"`
+	AggregationDelay  *int                     `json:"aggregationDelay,omitempty"`
+	AggregationTimer  *int                     `json:"aggregationTimer,omitempty"`
+}
+
+// AlertsNrqlConditionCreateSignal - Configuration that defines the signal that the NRQL condition will use to evaluate for Create.
+// nolint:revive
+type AlertsNrqlConditionCreateSignal struct {
+	AggregationWindow *int                     `json:"aggregationWindow,omitempty"`
+	EvaluationOffset  *int                     `json:"evaluationOffset,omitempty"`
+	FillOption        *AlertsFillOption        `json:"fillOption"`
+	FillValue         *float64                 `json:"fillValue"`
+	AggregationMethod *AlertsAggregationMethod `json:"aggregationMethod,omitempty"`
+	AggregationDelay  *int                     `json:"aggregationDelay,omitempty"`
+	AggregationTimer  *int                     `json:"aggregationTimer,omitempty"`
+}
+
+// AlertsNrqlConditionUpdateSignal - Configuration that defines the signal that the NRQL condition will use to evaluate for Update.
+// nolint:revive
+type AlertsNrqlConditionUpdateSignal struct {
+	AggregationWindow *int                     `json:"aggregationWindow,omitempty"`
+	EvaluationOffset  *int                     `json:"evaluationOffset,omitempty"`
+	FillOption        *AlertsFillOption        `json:"fillOption"`
+	FillValue         *float64                 `json:"fillValue"`
+	AggregationMethod *AlertsAggregationMethod `json:"aggregationMethod"`
+	AggregationDelay  *int                     `json:"aggregationDelay"`
+	AggregationTimer  *int                     `json:"aggregationTimer"`
+}
```

Comment:
I chose to create new Create and Update structs to allow the new Streaming Methods fields `AggregationMethod`, `AggregationDelay`,  `AggregationTimer`, which are optional fields, to be nulled out when a condition is updated.